utilities/vmware_guest_operations.py

```
@@ -0,0 +1,249 @@
+        GuestCommandError: If the command exits with a non-zero exit code
+        ValueError: If guest OS family cannot be determined or is unsupported
+        TimeoutExpiredError: If the command does not complete within timeout
+        vim.fault.GuestOperationsFault: If guest operations fail (auth, temp file, process)
+        urllib.error.URLError: If file transfer from guest fails
+    """
+    pm = content.guestOperationsManager.processManager
+    fm = content.guestOperationsManager.fileManager
+
+    guest_family = getattr(vm.guest, "guestFamily", None)
+    if not guest_family:
+        raise ValueError(f"Cannot determine guest OS for VM {vm.name}. Ensure VMware Tools is installed and running.")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)